datalad/interface/run_procedure.py

```diff
@@ -47,26 +49,64 @@ def _get_file_match(dir, name='*'):
     for target in targets:
         for m in iglob(op.join(dir, target)):
             m_bn = op.basename(m)
-            if name == '*' or m_bn == name or m_bn.startswith('{}.'.format(name)):
-                yield m
+            if name == '*':
+                report_name = m_bn[:-3] if m_bn.endswith('.py') or \
+                                           m_bn.endswith('.sh') \
+                                        else m_bn
+                yield m, report_name
+            elif m_bn == name or m_bn.startswith('{}.'.format(name)):
+                yield m, name
+
+
+def _get_proc_config(name, ds=None):
+    """get configuration of named procedure
+
+    Figures call format string and help message for a given procedure name,
+    based on dataset.
+
+    Returns
+    -------
+    tuple
+      (call format string, help string) or possibly None for either value,
+      if there's nothing configured
+    """
+    # figure what ConfigManager to ask
+    cm = cfg if ds is None else ds.config
+    v = cm.get('datalad.procedures.{}.call-format'.format(name), None)
+    h = cm.get('datalad.procedures.{}.help'.format(name), None)
+    if isinstance(v, tuple):
+        # If multiple values are defined (at the same level), ConfigManager
+        # returns a tuple. Go with the last one.
+        # TODO: Should we issue a warning or fail entirely?
```

Comment:
I don't think always returning a single value will be the solution. `ConfigManager` is explicitly made to be able to deal with multiple definitions of the same config item at the same level. But we probably should somehow be able to tell `ConfigManager` that for a particular item we only accept one value. However, I agree: This exceeds this PR.

Reply:
> I don't think always returning a single value will be the solution

Yes, you're right, things aren't so simple and some callers do expect multiple values.

> But we probably should somehow be able to tell ConfigManager that for a particular item we only accept one value.

Sounds good.  Based on frequency of use by our current callers, I wonder if "single value" should be the default and we should have a separate method for multiple items.

Anyway, I do have a question related to this PR:  Your comment says "If multiple values are defined (at the same level), ConfigManager returns a tuple".  I'm confused about what the "level" is here.   I don't think it is related to config location (global, .git/config, .datalad/config,...) because settings don't need to be at the same level to be represented as a tuple:

```
$ git config --global datalad.testconfig false                                                  
$ git config datalad.testconfig false                                                          
$ python -c "import datalad.api as dl; print(dl.Dataset('.').config.get('datalad.testconfig'))"
('false', 'false')                 
```

So the only way I can think of to interpret "at the same level" as meaning "this level in the dataset hierarchy".

P.S. I like this example because a naive caller would get say `if config.get(...)` and unexpectedly get something that evaluates to true even though both the global and local setting is false.  `getbool` also fails to handle it:

```
$ python -c "import datalad.api as dl; print(dl.Dataset('.').config.getbool('datalad', 'testconfig'))"
Traceback (most recent call last):                                                                                                     
  File "<string>", line 1, in <module>
  File "/home/kyle/src/python/datalad/datalad/config.py", line 473, in getbool
    return anything2bool(val)
  File "/home/kyle/src/python/datalad/datalad/config.py", line 113, in anything2bool
    % repr(val))
TypeError: Got value ('false', 'false') which could not be interpreted as a boolean
```

Reply:
> I don't think it is related to config location (global, .git/config, .datalad/config,...)

It is related. What I mean by "level" is the hierarchy of things we look at to find the (or a) procedure. In your case it's both "user level": First goes into `~/.gitconfig`, second into `.git/config`. Other levels would be "system", "dataset" (`.datalad/config`, which comes with the dataset in opposition to `.git/config`), datalad itself and last any datalad extension.

Re your PS:
That's a good catch and definitely sth to fix!

Reply:
Just for clarification: Those are the levels to be searched for procedures by `run-procedure` and also the levels of consideration by `ConfigManager` when figuring what to return. And this again is related to the change of order you noticed. They are consistent now. (But I still should write down the rational for that particular order)

Reply:
Thanks for clarifying. I think your mapping between "procedure levels" and "config levels" breaks down because AFAICS `ConfigManager` (as a getter) only distinguishes two levels: dataset (".datalad/config") versus "everything else git config will report".

If I have a setting in `/etc/gitconfig`, `~/.gitconfig`, and `$REPO/.git/config`, `ConfigManager` reports

```shell
$ python -c "import datalad.api as dl; print(dl.Dataset('.').config.get('datalad.test'))"
('etc-config', 'home-config', 'repo-config')
```

By procedure level standards, that includes both "system" and "user" levels.

Reply:
True, but this reporting by `git-config` still happens in a particular order (more general to more specific). So, this is the reason I went with the last entry in case of multiple values returned.

I'll try to articulate this idea more clearly in comment/docstring